Fix CI

Fancy2209 opened the pull request:
I'm sorry for the issues
- Remove macOS14 Runner (LibreSSL is conflicting with OpenSSL on build)
- Disable macOS 12 Unit Tests (I cannot fix this as I do not have a mac or the necessary knowledge)
- Fix the Unit Tests Command on Windows
- Fix Install Paths for MonaBase and MonaCore
- Fix Linux Binaries the libs to the right place

Fancy2209 commented:
I had to disable UnitTests on mac, who isn't good because I've had a person have issues on Mac, I unfortunately have no idea why they weren't working, will try to fix them on a separate PR

Fancy2209 commented:
OH
it passes CI now, it just doesn't show logs

Fancy2209 commented:
YES!
It works now

Fancy2209 commented:
I asked someone to test
It currently isn't statically linked so it won't work on Mac without a manual build

Fancy2209 commented:
Same issue on linux
./MonaServer: error while loading shared libraries: libMonaBase.so: cannot open shared object file: No such file or directory
It'll search where the library was located when it was built

thomasjammet commented:
Hi Fancy, thanks for the fixes!

It just misses the `../MonaBase/lib/libMonaBase.so` and `../MonaCore/lib/libMonaCore.so` libraries, can you include it inside the artifact? Same problem for MacOS

LibreSSL is conflicting with OpenSSL => something to check so, I guess everyone will get the issue.

Fancy2209 commented:
> Hi Fancy, thanks for the fixes!
> 
> It just misses the `../MonaBase/lib/libMonaBase.so` and `../MonaCore/lib/libMonaCore.so` libraries, can you include it inside the artifact? Same problem for MacOS
> 
> LibreSSL is conflicting with OpenSSL => something to check so, I guess everyone will get the issue.

It's only happening on MacOS 14
Also I think the macOS Port just is not working properly too. I've had someone try to host a RTMFP Server and connect and it failed

Fancy2209 commented:
This is done! (for now)
I do not have a mac to fix the macOS Unit Tests
thanks @bugaevc for telling me how to fix the macOS Build!

MathieuPOUX commented:
Thanks really for your contribution @Fancy2209 !! It looks me good to merge, good for you too @thomasjammet ?

thomasjammet commented on MonaServer/lib/libMonaBase.a:
it shouldn't be in git, can you remove it please?

thomasjammet commented on MonaServer/lib/libMonaBase.so:
it shouldn't be in git, can you remove it please?

thomasjammet commented on MonaServer/lib/libMonaCore.so:
it shouldn't be in git, can you remove it please?

thomasjammet commented:
> Thanks really for your contribution @Fancy2209 !! It looks me good to merge, good for you too @thomasjammet ?

Thanks Fancy, I think it's ok, can you just remove linux binaries from git?

Fancy2209 commented:
> > Thanks really for your contribution @Fancy2209 !! It looks me good to merge, good for you too @thomasjammet ?
> 
> Thanks Fancy, I think it's ok, can you just remove linux binaries from git?

I thought I hadn't committed those, ups!
Will fix asap

Fancy2209 commented:
I had changed
```          mkdir -p MonaBin/MonaServer
          mkdir -p MonaBin/MonaBase/lib
          mkdir -p MonaBin/MonaCore/lib
```
To
```          mkdir -p MonaBin/MonaServer/bin
```
Because I changed the dylib path to be like Linux, who is a lib folder next to the executable, any reason to change this?

Fancy2209 commented:
Also from my testing tbe rpath needed no changes, it was working in Linux and everything indicated Darwin too

Fancy2209 commented:
If you prefer the 3 folder way the install path needs changes, not rpath

thomasjammet commented:
Thanks for the last fixes! I can't tell about the macos part because unfortunately I don't have a way to test it right now.

About linux part it was not working, it is something I already experienced, by default RPATH is changed to RUNPATH and is using absolute path. You can check with :
`objdump -p ./MonaServer`

Fancy2209 commented:
> Thanks for the last fixes! I can't tell about the macos part because unfortunately I don't have a way to test it right now.
> 
> About linux part it was not working, it is something I already experienced, by default RPATH is changed to RUNPATH and is using absolute path. You can check with : `objdump -p ./MonaServer`

Ah thanks, I also don't have a mac btw, I was using darlinghq to check if the dylibs loaded
